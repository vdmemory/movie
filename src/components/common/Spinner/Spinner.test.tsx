import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import Spiner from "./Spinner";

const setup = () => {
  const renderer = createRenderer();
  renderer.render(<Spiner />);
  const output = renderer.getRenderOutput();
  return output;
};

describe("Spinner Component", () => {
  test("Should renders correctly", () => {
    const output = setup();
    expect(output).toMatchSnapshot();
  });

  test("Should render wrapSpinner class", () => {
    const output = setup();
    expect(output.props.className).toBe("wrapSpinner");
  });

  test("Should render spinner class", () => {
    const output = setup();
    const spinner = output.props.children;
    expect(spinner.props.className).toBe("spinner");
  });
});
