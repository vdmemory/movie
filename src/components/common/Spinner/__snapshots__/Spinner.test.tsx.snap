// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component Should renders correctly 1`] = `
<div
  className="wrapSpinner"
>
  <div
    className="spinner"
  />
</div>
`;
